parsons/ngpvan/people.py

```diff
@@ -146,23 +146,58 @@ def _people_search(self, first_name=None, last_name=None, date_of_birth=None, em
     def _valid_search(self, first_name, last_name, email, phone, dob, street_number,
                       street_name, zip, match_map):
         # Internal method to check if a search is valid
-
-        if (None in [first_name, last_name, email] and
-            None in [first_name, last_name, phone] and
-            None in [first_name, last_name, zip, dob] and
-            None in [first_name, last_name, street_number, street_name, zip] and
-                None in [email]):
-
-            raise ValueError("""
-                             Person find must include the following minimum
-                             combinations to conduct a search.
-                                - first_name, last_name, email
-                                - first_name, last_name, phone
-                                - first_name, last_name, zip, dob
-                                - first_name, last_name, street_number, street_name, zip
-                                - email
-                            """)
-
+        if not match_map:
```

Comment:
I _think_ the best way to accomplish this would actually be to hook onto the logic already present in `_people_search()`. My instinct is that we should call `_valid_search()` towards the end of `_people_search()`, and just pass the variable `json`, which will either be a collection of the individual field arguments, or, if provided, `match_map`. Then `_valid_search()` would just need to check the `json` variable. The main down-side to this approach is that we'd need to flatten `json` to get at all the values, but I think it would still save many lines of code and would be overall much more readable (though I could certainly be convinced otherwise).

Reply:
I like that approach, should we even give the option for match map? The user would have to know exactly how people objects are structured in VAN, if we do receive a match map maybe we can have _people_search check the structure and throw an error if it's not in the right format. For example if zipOrPostalCode is not in the addresses value.

Reply:
That sounds like a bunch of extra work, but as someone who's had to use `match_map` I know I'd be grateful for the extra checking! It does seem like the checking ideally _should_ happen in `_valid_search()`, but that's mostly for thematic consistency. If it's much simpler to put that into `_people_search()` I certainly wouldn't complain.
I just made sure that the `vanid_endpoints` branch is still available here on Github. There were problems with the way I tried to handle this issue (which may be tangled because that PR tried to fix too many things at once), but on the off chance my code can be of any help to you: https://github.com/move-coop/parsons/blob/vanid_endpoints/parsons/ngpvan/people.py